### ISSUE TITLE
[SUGGESTION] Request for option to specify RuntimeIdentifier for iOS Simulator when debugging in .Net MAUI

### ISSUE THREAD

**Issue by xanril** [microsoft/vscode-dotnettools#287]
Type: <b>Feature Request</b>

I am currently trying to debug an existing project built from VS For Mac 2022 into VS Code with .Net MAUI plugin. I've selected an iOS Simulator as a debug target. However, debugging using the built-in prelaunch task `maui: Build` fails.

I noticed that when VS Code builds for debug, it uses `-r iossimulator-arm64` in its build command by default. This causes most of my iOS NuGet references to fail, since it they only have binaries for `iossimulator-x64`. 

I also observed that VS For Mac 2022 builds/compiles the app using this runtime identifier by default when I have a simulator as debug target, which makes it run without any issues. 

I tried running the same build command in the VS Code terminal after changing the `RuntimeIdentifier` to `iossimulator-x64`, it builds fine. I also did try to create a custom vs code task which runs this command instead of the built-in `maui: Build`. It builds fine, but fails to find/run the mlaunch task after the build process.

Would it be possible to override the default `RuntimeIdentifer` in `maui: Build` task so that I can specify the simulator runtime compatible for my project?

Extension version: 0.1.34
VS Code version: Code 1.80.0 (660393deaaa6d1996740ff4880f1bad43768c814, 2023-07-04T13:40:17.963Z)
OS version: Darwin arm64 22.4.0
Modes:

**Comment by edgedj** [comment]
Has anyone come up with a workaround for this or is it being applied as a fix. At present it's not possible to run a Maui app on iOS from VS Code if Microsoft.AppCenter.Analytics is included, the standard sample project works ok but just referencing that package breaks it when running on Apple Silicon from VS Code.

**Comment by flipper09112** [comment]
+1

**Comment by quiest2000** [comment]
take a look at [this comment ](https://github.com/microsoft/vscode-dotnettools/issues/744#issuecomment-1889797200)

**Comment by xanril** [comment]
Thanks @quiest2000! 

I confirmed that the changes from that bug solved this suggestion. I was able to properly deploy and debug projects that target `iossimulator-x64` either by adding `<ForceSimulatorX64ArchitectureInIDE>true</ForceSimulatorX64ArchitectureInIDE>` or `<RuntimeIdentifier>iossimulator-x64</RuntimeIdentifier>` property in `.csproj`.

Hoping this does not break in the future. Closing this suggestion.
